### PR TITLE
chore: rebuild bundle if React is disabled

### PR DESCRIPTION
**Pull request by tltv**
Default production and development bundles are packaged with react router only. With this change, bundle is always rebuild if React is disabled and default bundle would be used otherwise. Rebuilding ensures that Vaadin router is used instead of the React router.

Fixes: #18656

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/BundleValidationTest.java [hunk 1]
```diff
@@ -2121,6 +2152,37 @@ public void defaultProdBundleExists_noCompressedProdBundleFile_noBuildRequired()
                 needsBuild);
     }
 
+    @Test
+    public void defaultProdBundleExists_noCompressedProdBundleFile_reactDisabled_noBuildRequired()
+            throws IOException {
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/BundleValidationTest.java:2157]:
```suggestion
    public void defaultProdBundleExists_noCompressedProdBundleFile_reactDisabled_buildRequired()
            throws IOException {
```
should be build required. Perhaps `prodBuildRequired`

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java [hunk 2]
```diff
@@ -135,6 +137,15 @@ private static boolean needsBuildDevBundle(Options options,
             return false;
         }
 
+        if (!DevBundleUtils
+                .getDevBundleFolder(npmFolder, options.getBuildDirectoryName())
+                .exists() && !options.isReactEnabled()) {
+            // Using default dev bundle in a Jar.
+            // Default dev bundle is packaged with react router only. With react
+            // disabled, let's rebuild bundle to get vaadin router instead.
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:145]: could have an info log that `Bundle build required for non default router.` or something like that

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java [hunk 3]
```diff
@@ -152,6 +163,15 @@ private static boolean needsBuildDevBundle(Options options,
     private static boolean needsBuildProdBundle(Options options,
             FrontendDependenciesScanner frontendDependencies)
             throws IOException {
+
+        if (!options.isReactEnabled() && !ProdBundleUtils
+                .getProdBundle(options.getNpmFolder()).exists()) {
+            // Using default prod bundle in a Jar.
+            // Default prod bundle is packaged with react router only. With
+            // react disabled, let's rebuild bundle to get vaadin router
+            // instead.
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:172]: could have an info log that `Bundle build required for non default router.` or something like that

**Review by caalador — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java (hunk 2 above)
Comment (tltv) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:145]: done

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java (hunk 3 above)
Comment (tltv) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:173]: done

**Review by caalador — APPROVED** [review]

**Pull request merged**